### PR TITLE
Adds concat_uvp to combine multiple UVPSpec objects

### PR DESCRIPTION
**Pull request by nkern**
Adds `uvpspec.concat_uvp`, as well as reformats some of the meta-data in a `UVPSpec` obj in order to do so. Also adds `testing.py` w/ functions to build example `UVPSpec` objects

### CODE REVIEW

File: hera_pspec/uvpspec.py [hunk 1]
```diff
@@ -1374,3 +1433,305 @@ def compute_scalar(self, spw, pol, num_steps=1000, little_h=True,
...
+    find a single data axis ['spw', 'blpairts', 'pol'] where *all* objects contain 
+    completely non-overlapping data. Overlapping data are delay spectra that
+    have identical spw, blpair-time and pol metadata between each other.
+
+    Parameters
+    ----------
+    uvps : list of UVPSpec objects or list of string paths to UVPSpec objects
+
+    just_meta : boolean, optional
+        If uvps contain list of strings, load only the metadata in to get uvp overlap.
```
Comment (philbull) [inline hera_pspec/uvpspec.py]: Not sure what this means.

File: hera_pspec/uvpspec.py [hunk 2]
```diff
@@ -1374,3 +1433,305 @@ def compute_scalar(self, spw, pol, num_steps=1000, little_h=True,
...
+
+    return u
+
+
+def get_uvp_overlap(uvps, just_meta=True, verbose=True):
+    """
+    Given a list of UVPSpec objects or a list of paths to UVPSpec objects,
+    find a single data axis ['spw', 'blpairts', 'pol'] where *all* objects contain 
+    completely non-overlapping data. Overlapping data are delay spectra that
+    have identical spw, blpair-time and pol metadata between each other.
```
Comment (philbull) [inline hera_pspec/uvpspec.py]: Maybe mention that an error is raised if there are no non-overlapping axes?

File: hera_pspec/pspecdata.py [hunk 3]
```diff
@@ -1465,14 +1468,28 @@ def pspec(self, bls1, bls2, dsets, pols, input_data_weight='identity', norm='I',
...
+        uvp.labels = sorted(set([label1, label2]))
+        uvp.label_1_array = np.ones((uvp.Nspws, uvp.Nblpairts, uvp.Npols), np.int) \
+                            * uvp.labels.index(label1)
+        uvp.label_2_array = np.ones((uvp.Nspws, uvp.Nblpairts, uvp.Npols), np.int) \
+                            * uvp.labels.index(label2)
+        uvp.labels = np.array(uvp.labels, np.str)
+        uvp.history = "UVPSpec written on {} with hera_pspec git hash {}\n{}\n" \
+                      "dataset1: filename: {}, label: {}, history:\n{}\n{}\n" \
+                      "dataset2: filename: {}, label: {}, history:\n{}\n{}\n" \
+                      "".format(datetime.datetime.now(), version.git_hash, '-'*20,
```
[the remark this answers is not in the document]
Comment (nkern) [reply hera_pspec/pspecdata.py]: good point

File: hera_pspec/testing.py [hunk 4]
```diff
@@ -0,0 +1,163 @@
...
+from pyuvdata import UVData
+
+
+def build_vanilla_uvpspec(beam=None):
+    """
+    Build an example vanilla UVPSpec object from scratch, with all necessary metadata.
+
+    Parameters
+    ----------
+    beam : PSpecBeamUV object
```
[the remark this answers is not in the document]
Comment (nkern) [reply hera_pspec/testing.py]: nope, just any subclass of `PspecBeamBase`. I updated that in the docstring, thanks

File: hera_pspec/testing.py [hunk 5]
```diff
@@ -0,0 +1,163 @@
...
+    label1 = 'red'
+    label2 = 'blue'
+    labels = np.array([label1, label2])
+    label_1_array = np.ones((Nspws, Nblpairts, Npols), np.int) * 0
+    label_2_array = np.ones((Nspws, Nblpairts, Npols), np.int) * 1
+    if beam is not None:
+        OmegaP, OmegaPP = beam.get_Omegas(beam.primary_beam.polarization_array[0])
+        beam_freqs = beam.beam_freqs
+
+    telescope_location = np.array([5109325.85521063,
```
[the remark this answers is not in the document]
Comment (nkern) [reply hera_pspec/testing.py]: sure thing

File: hera_pspec/testing.py [hunk 6]
```diff
@@ -0,0 +1,163 @@
...
+    
+    # Set all parameters
+    for p in params:
+        setattr(uvp, p, locals()[p])
+    
+    uvp.check()
+
+    return uvp, cosmo
+
+def build_uvpspec_from_data(data, bls, spw_ranges=None, beam=None, taper='none', cosmo=None, verbose=False):
```
[the remark this answers is not in the document]
Comment (nkern) [reply hera_pspec/testing.py]: sure

File: hera_pspec/testing.py [hunk 7]
```diff
@@ -0,0 +1,163 @@
...
+    ----------
+    data : UVData object or path to miriad file
+
+    bls : list of at least 2 baseline tuples
+        Ex: [(24, 25), (37, 38), ...]
+
+    spw_ranges : list of spw tuples
+        See PSpecData.pspec docstring
+
+    beam : PSpecBeamUV object or path to beamfits healpix beam
```
[the remark this answers is not in the document]
Comment (nkern) [reply hera_pspec/testing.py]: okay, thanks for the heads-up

File: hera_pspec/uvpspec.py [hunk 8]
```diff
@@ -1374,3 +1433,305 @@ def compute_scalar(self, spw, pol, num_steps=1000, little_h=True,
         return scalar
 
 
+def concate_uvp(uvps, verbose=True):
```
[the remark this answers is not in the document]
Comment (nkern) [reply hera_pspec/uvpspec.py]: sure

File: hera_pspec/uvpspec.py (hunk 1 above)
Comment (nkern) [reply hera_pspec/uvpspec.py]: clarified doc-string, lmk if this helps

**Pull request merged**